tests/test_signer.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
+
+"""Test cases for 'signer.py'. """
+
+import sys
+import unittest
+
+import unittest
+import securesystemslib.formats
+import securesystemslib.keys as KEYS
+from securesystemslib.exceptions import FormatError, UnsupportedAlgorithmError
+
+# TODO: Remove case handling when fully dropping support for versions >= 3.6
+IS_PY_VERSION_SUPPORTED = sys.version_info >= (3, 6)
+
+# Use setUpModule to tell unittest runner to skip this test module gracefully.
+def setUpModule():
+    if not IS_PY_VERSION_SUPPORTED:
+        raise unittest.SkipTest('requires Python 3.6 or higher')
+
+# Since setUpModule is called after imports we need to import conditionally.
+if IS_PY_VERSION_SUPPORTED:
+    from securesystemslib.signer import Signature, SSlibSigner
+
+
+class TestSSlibSigner(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.rsakey_dict = KEYS.generate_rsa_key()
+        cls.ed25519key_dict = KEYS.generate_ed25519_key()
+        cls.ecdsakey_dict = KEYS.generate_ecdsa_key()
+        cls.DATA_STR = 'SOME DATA REQUIRING AUTHENTICITY.'
+        cls.DATA = securesystemslib.formats.encode_canonical(
+                cls.DATA_STR).encode('utf-8')
+
+
+    def test_sslib_sign(self):
+        dicts = [self.rsakey_dict, self.ecdsakey_dict, self.ed25519key_dict]
+        for scheme_dict in dicts:
+            # Test generation of signatures.
+            sslib_signer = SSlibSigner(scheme_dict)
+            sig_obj = sslib_signer.sign(self.DATA)
+
+            # Verify signature
+            verified = KEYS.verify_signature(scheme_dict, sig_obj.to_dict(),
+                    self.DATA)
+            self.assertTrue(verified, "Incorrect signature.")
+
+            # Removing private key from 'scheme_dict' - should raise a ValueError.
+            private = scheme_dict['keyval']['private']
+            scheme_dict['keyval']['private'] = ''
+            sslib_signer.key_dict = scheme_dict
+
+            with self.assertRaises((ValueError, FormatError)):
+                sslib_signer.sign(self.DATA)
+
+            scheme_dict['keyval']['private'] = private
+
+            # Test for invalid signature scheme.
+            valid_scheme = scheme_dict['scheme']
+            scheme_dict['scheme'] = 'invalid_scheme'
+            sslib_signer = SSlibSigner(scheme_dict)
+
+            with self.assertRaises((UnsupportedAlgorithmError, FormatError)):
```

Comment:
Is this also different for different keys? 🤦

Reply:
Yes, for `rsa` we have a check in `securesystemslib.keys.create_signature()` which we don't have in for `ed25519` and `ecdsa`:
![image](https://user-images.githubusercontent.com/16246778/108843900-fe55e480-75e3-11eb-92a9-ef864d581da8.png)

but when we call `ed25519_keys.create_signature` or `ecdsa_keys.create_signature` there is a check inside those
functions for the `scheme` argument and that's how a FormatError is received. 

Do you think I should fix those problems here in this or?
Or in a different pr or issue?
Or even should I fix it now given that I am investigating how to replace the schemas?

Reply:
Thanks for the detailed report! Let's just add this to the wishlist in #191.